File: src/connect-k-python/GameLogic.py
from BoardClasses import *
from ManualAI import *
from StudentAI import *

class GameLogic:

    def __init__(self,col,row,k,g,mode,debug):
        self.col = col
        self.row = row
        self.k = k
        self.g = g
        self.mode = mode
        self.debug = debug
        self.ai_list = []

    def Manual(self):
        player = 1
        winPlayer = 0
        move = Move(-1,-1)
        board = Board(self.col,self.row,self.k,self.g)
        while True:
            move = self.ai_list[player-1].get_move(move)
            try:
                board = board.make_move(move,player)
            except InvalidMoveError:
                print("Invalid Move!")
                if player == 1:
                    winPlayer = 2
                else:
                    winPlayer = 1
                break
            winPlayer = board.is_win()
            board.show_board()
            if(winPlayer != 0):
                break
            if player == 1:
                player = 2
            else:
                player = 1
        if winPlayer == -1:
            print("Tie")
        else:
            print('player',winPlayer,'wins')

    def TournamentInterface(self):
        ai = StudentAI(self.col,self.row,self.k,self.g)
        while True:
            col, row = map(lambda x: int(x), input().split(' '))
            result = ai.get_move(Move(col,row))
            print("{} {}".format(result.col,result.row))


    def Run(self):
        if self.mode == 'm':
            self.ai_list.append(ManualAI(self.col, self.row, self.k, self.g))
            self.ai_list.append(StudentAI(self.col,self.row,self.k,self.g))

            self.Manual()
        if self.mode == 't':
            self.TournamentInterface()

File: src/connect-k-java/GameLogic.java
import java.util.*;
import java.io.*;
import java.lang.*;

public class GameLogic {
	private int col, row, k ,g;
	private String mode;
	private boolean debug;
	private ArrayList<AI> aiList;

	public GameLogic(int col, int row, int k, int g, String mode, boolean debug) {
		this.col = col;
		this.row = row;
		this.k = k;
		this.g = g;
		this.mode = mode;
		this.debug = debug;
		aiList = new ArrayList<AI>();
	}

	public void Manual() 
	{
		int player = 1;
		int winPlayer = 0;
		boolean init = true;
		Move move = new Move(-1,-1);
		Board board = new Board(col,row,k,g);
		while (true)
		{
			move = aiList.get(player - 1).GetMove(move);
			try
			{
				board = board.MakeMove(move, player);
			}
			catch (InvalidMoveError e)
			{
				winPlayer = player == 1 ? 2 : 1;
				break;
			}

			winPlayer = board.IsWin();
			board.ShowBoard();
			if (winPlayer != 0)
			{
				break;
			}
			player = player == 1 ? 2 : 1;

		}
		if (winPlayer == -1)
		{
		    System.out.println("Tie");
		}
		else
		{
		    System.out.println("Player " + winPlayer + " wins!");
		}

	}

	public void TournamentInterface() 
	{
		Scanner reader = new Scanner(System.in);
		StudentAI ai = new StudentAI(col, row, k, g);
		while (true) 
		{
			int col , row;
			col = reader.nextInt();
			row = reader.nextInt();
			Move result = ai.GetMove(new Move(col, row));
			System.out.println(result.col + " " + result.row + " ");
		}
	}

	public void Run() 
	{
		if ( this.mode.equals("m") )
		{
			AI studentai = new StudentAI(col, row, k, g);
			AI manualai = new ManualAI(col, row, k, g);
			aiList.add(studentai);
			aiList.add(manualai);
			Manual();
		}

		else if (mode.equals("t") )
		{
			TournamentInterface();
		}

	}
}